### DISCUSSION TITLE
Using with Livewire

### DISCUSSION THREAD

**Discussion by seabasss** [danharrin/alpine-mousetrap#5] [General]
Hey, it's me again ;)

I'm trying to use this to submit a form with Livewire lazy fields and when I run my keycommand to save the form it saves it before the lazy update triggers. Any tips on how to best get around this?

I have this on my save button `x-mousetrap.global.ctrl+s`

I could switch to debounce, but would love to keep lazy and even with debounce I guess there would be a possibility of saving the form before the form update happens.

Thanks for tips!

**Comment by danharrin** [comment]
Hey! Maybe, on save, try using JS to search for all inputs with the wire:model.lazy attribute, and dispatch a `change` event on them? Then they will be updated before you save

Otherwise, switch to defer?

**Reply by seabasss** [reply]
Ah, thanks! I have a few fields that need to lazy update unfortunately.

Do you mean trigger JS from the key command instead of triggering submit button click? Or keep the key command as is and run JS on submit click/trigger?

If the latter, would I put JS in an x-on:click on the button and hope that it fires before wire:click?

Technically all I have to do is to move focus/blur and lazy update will happen?

Thank you so much!

**Reply by danharrin** [reply]
```html
x-mousetrap.global.ctrl-s="
    document.querySelectorAll('input[wire:model.lazy]')
        .forEach((input) => input.dispatchEvent(new Event('change')))

    $el.click()
"
```

Maybe? Unless the lazy updates are too slow and the click gets triggered first

**Reply by seabasss** [reply]
Thank you!

I got an error saying `Alpine Expression Error: Document.querySelectorAll: 'input[wire:model.lazy]' is not a valid selector`, but with your help I think I came up with a different solution.

Any downsides that you can see from doing the following?

```
x-mousetrap.global.ctrl-s="
    document.activeElement.blur()

    $el.click()
"

Big thanks again!

**Reply by danharrin** [reply]
Eh, technically you could just do `input` and properly emit the change events. But does the blur work?

**Reply by seabasss** [reply]
Yeah the blur seems to work and I only really need to make sure that the currently focused element gets updated! Wondered if it would be a lighter operation than iterate through every single input on the page. I have quite a few :)

**Reply by danharrin** [reply]
Yeah that sounds good, as long as it works haha
